scripts/step-3.sh

```diff
@@ -44,11 +44,15 @@ clean_state
 checkout_branch
 
 # Snapshot and sign the snapshot with snapshot kms key
-./tuf snapshot -repository "$REPO"
-./tuf sign -repository "$REPO" -roles snapshot -key "${SNAPSHOT_KEY}"
+if [ -n "$SNAPSHOT" ]; then
+    ./tuf snapshot -repository "$REPO"
+    ./tuf sign -repository "$REPO" -roles snapshot -key "${SNAPSHOT_KEY}"
+fi
 
 # Timestamp and sign the timestamp with timestamp kms key
-./tuf timestamp -repository "$REPO"
-./tuf sign -repository "$REPO" -roles timestamp -key "${TIMESTAMP_KEY}"
+if [ -n "$TIMESTAMP" ]; then
```

Comment:
here's probably the best place to handle that timestamp must run if either snapshot or timestamp is defined.